Es dpi

jvelilla opened the pull request.

jvelilla committed:
Update .travis.yml

jvelilla committed:
TravisCI with MacOS

jvelilla committed:
Merge branch 'master' of https://github.com/jvelilla/EiffelStudio

# Conflicts:
#	.travis.yml

jvelilla committed:
Update .travis.yml

jvelilla committed:
Update .travis.yml

Compile only unstable library

jvelilla committed:
Update .travis.yml

jvelilla committed:
Update .travis.yml

jvelilla committed:
Update .travis.yml

jvelilla committed:
Merge branch 'master' of https://github.com/EiffelSoftware/EiffelStudio

# Conflicts:
#	.travis.yml

jvelilla committed:
Updated HighDPI support on Windows.
Handled WM_DPICHANGED, added a new event EV_DPI_ACTION_SEQUENCE to handle
dpi change on any Widget.

jvelilla committed:
Updated EV_WIDGET_IMP with feature `init_dpi_changed_actions` for GTK implementation.

jvelilla committed:
Updated cocoa and gtk implementations added an empty implementation for
`init_dpi_changed_actions` feature.

Updated EV_MONITOR_DPI_DETECTOR with a new feature to compute the scale value
for the new dpi.

jocelyn commented on .travis.yml:
```diff
@@ -2,23 +2,35 @@
 - export current_dir=$PWD ; echo current_dir=$current_dir ; cd ..
-- curl -sSL https://www.eiffel.org/setup/install.sh | bash -s nightly > eiffel.rc
-- source ./eiffel.rc
+- export ISE_EIFFEL=$PWD/Eiffel_18.11;
```
do not commit this.

jocelyn commented on Src/Eiffel/interface/graphical/foundations/tools/es_dockable_tool_panel.e:
```diff
@@ -1004,7 +1017,7 @@ feature {NONE} -- Factory
 
             create l_container
-            l_container.set_minimum_height (26)
+            l_container.set_minimum_height ((create {EV_MONITOR_DPI_DETECTOR_IMP}).scaled_size (26))
```
If it is possible to have instance-free call, this would be better, otherwise ok.

jvelilla replied:
I will check.

jocelyn commented on Src/Eiffel/interface/graphical/tools/testing/widget/es_test_statistics_widget.e:
```diff
@@ -255,6 +256,13 @@ feature {NONE} -- Implementation
+		do
+			redraw_status_bar (a_x, a_y, a_width, a_height)
+		end
+
```
remove one empty line

jvelilla replied:
ok

jocelyn commented on Src/Eiffel/interface/new_graphical/dialogs/es_breakpoint_dialog.e:
```diff
@@ -725,6 +725,17 @@ feature {NONE} -- When hits
 				)
+			l_scroll.dpi_changed_actions.extend (agent (asc: EV_SCROLLABLE_AREA; a_dpi,ax,ay,aw,ah: INTEGER)
+					do
+						asc.set_item_size (asc.item.minimum_width.max (asc.width - vertical_scrollbar_width), asc.item.minimum_height.max (asc.height - horizontal_scrollbar_height))
```
I am wondering, if this code could not be moved to EV_SCROLLABLE_AREA under a new feature ?
maybe "update_size"  or similar. 
This way, you could reuse it for resize_actions on l_scroll.  This avoids code duplication, it would make the executable a (tiny) bit smaller, ...

jvelilla replied:
I see, but I don't know what to do with the extra argument for `dpi`, at the moment it's not used.

jocelyn commented on Src/framework/configuration/gui/dialogs/add_library_dialog.e:
```diff
@@ -105,6 +105,12 @@ feature {NONE} -- Initialization
 					a_split.set_proportion (0.9)
 				end (?, ?, ?, ?, hsp))
 
+			dpi_changed_actions.extend (agent (a_dpi, ia_x, ia_y, ia_width, ia_height: INTEGER; a_split: EV_VERTICAL_SPLIT_AREA)
```
Not sure this is really critical here.

jocelyn commented on Src/library/docking/implementation/controls/sd_horizontal_split_area.e:
```diff
@@ -50,6 +50,9 @@ feature {NONE} -- Implementation
 				(agent (a_x, a_y, a_width, a_height: INTEGER_32)
 					do remember_top_resize_split_area (Current) end)
+			dpi_changed_actions.extend
+					(agent (a_dpi,a_x, a_y, a_width, a_height: INTEGER_32)
```
you should keep same indentation as previous case with resize_actions ...

jvelilla replied:
Done

jocelyn commented on Src/library/docking/implementation/controls/sd_title_bar.e:
```diff
@@ -502,6 +503,76 @@ feature {NONE} -- Agents
 			end
 		end
 
+	on_dpi_change_fixed_resize (a_dpi, a_x: INTEGER_32; a_y: INTEGER_32; a_width: INTEGER_32; a_height: INTEGER_32)
```
anyway to factorise code with on_fixed_resize feature??

jocelyn replied:
And this one ?

jvelilla replied:
Done.

jocelyn commented on Src/library/docking/implementation/docking/sd_docking_manager_agents.e:
```diff
@@ -208,6 +209,57 @@ feature  -- Agents
 		end
 
+
+	on_dpi_change_resize (a_dpi,a_x: INTEGER; a_y: INTEGER; a_width: INTEGER; a_height: INTEGER; a_force: BOOLEAN)
```
I guess code could be factorized by on_resize ...

jocelyn commented on Src/library/docking/implementation/docking/sd_docking_manager_agents.e:
```diff
@@ -208,6 +209,57 @@ feature  -- Agents
+			l_docking_manager: like docking_manager
+		do
+			debug ("docking")
+				io.put_string ("%N SD_DOCKING_MANAGER on_resize ~~~~~~~~~~~~~~~~~~~~")
```
in manifest string : on_resize or on_dpi_change_resize ?

jocelyn replied:
and this one ?

jvelilla replied:
Done

jocelyn commented on Src/library/editor/text_window/interface/text_panel.e:
```diff
@@ -1326,6 +1337,34 @@ feature {NONE} -- Display functions
 			end
 		end
 
+	on_dpi_change_viewport_size (a_dpi, a_x, a_y: INTEGER; a_width, a_height: INTEGER)
```
factorize or reuse `on_viewport_size `?

jocelyn replied:
and that one?

jvelilla replied:
Done.

jocelyn commented on Src/library/memory_analyzer/widgets/ma_window.e:
```diff
@@ -176,12 +181,26 @@ feature {NONE} -- Implementation for agents
+	eiffel_view_frame_dpi_size_change (a_dpi, a_x, a_y, a_width, a_height: INTEGER)
+			-- Handle frame change.
+		do
+
```
even if it does nothing , I would call `eiffel_view_frame_size_change` 
this way, if it is redefined, or changed in the future ...

On the other hand, if the `eiffel_view_frame_size_change ` is empty .. could we remove it ?

jvelilla replied:
I will use the first option but I will add a TODO to check if it's really used.

jocelyn commented on Src/library/memory_analyzer/widgets/ma_window.e:
```diff
@@ -330,12 +349,28 @@ feature {NONE} -- Implementation for agents
+		require else
+			a_height_valid: a_height >= 0 and a_height <= 800
+		do
+			analyze_gc.resize (a_x, a_y, a_width, a_height)
```
call  `resize_histogram`

jvelilla replied:
Done

jocelyn commented on Src/library/memory_analyzer/widgets/ma_window.e:
```diff
@@ -330,12 +349,28 @@ feature {NONE} -- Implementation for agents
+	dpi_resize_history (a_dpi, a_x, a_y, a_width, a_height: INTEGER)
+			-- Do things when the eiffel_histogram size change.
+		do
+			analyze_gc.resize_history (a_x, a_y, a_width, a_height)
```
call `resize_history `

jvelilla replied:
Done

jocelyn commented on Src/library/memory_analyzer/widgets/ma_window_imp.e:
```diff
@@ -1026,11 +1030,21 @@ feature {NONE} -- Implementation
 
+	eiffel_view_frame_dpi_size_change (a_dpi, a_x, a_y, a_width, a_height: INTEGER)
+			-- Called by `dpi_changed_actions' of `eiffel_view_frame'.
+		deferred
```
Maybe here, no deferred, but effective feature calling ` eiffel_view_frame_size_change `

jvelilla replied:
yes, then the previous one is not needed for now.

jocelyn commented on Src/library/memory_analyzer/widgets/ma_window_imp.e:
```diff
@@ -1041,6 +1055,11 @@ feature {NONE} -- Implementation
 
+	dpi_resize_history (a_dpi, a_x, a_y, a_width, a_height: INTEGER)
+			-- Called by `dpi_changed_actions' of `eiffel_history'.
+		deferred
```
maybe , you should have an effective feature calling `resize_history `

jvelilla replied:
Done.

jocelyn commented on Src/library/vision2/implementation/implementation_interface/properties/ev_positioned_i.e:
```diff
@@ -58,6 +58,12 @@ feature -- Measurement
 		deferred
 		end
 
+	dpi: INTEGER
```
Why INTEGER and not NATURAL as in EV_MONITOR_DPI_DETECTOR_IMP ?

jocelyn replied:
or the reverse


Why NATURAL in EV_MONITOR_DPI_DETECTOR_IMP and not INTEGER ?

jvelilla replied:
Done updated to NATURAL.

jocelyn commented on Src/library/vision2/implementation/implementation_interface/widgets/primitives/ev_grid_i.e:
```diff
@@ -4854,6 +4855,43 @@ feature {EV_GRID_LOCKED_I} -- Drawing implementation
+			a_width_non_negative: a_width >= 0
+			a_height_non_negative: a_height >= 0
+		do
+				-- Set the internal client dimensions for
```
reuse viewport_resized  ?

jocelyn commented on Src/library/vision2/implementation/mswin/support/ev_wel_control_container_imp.e:
```diff
@@ -223,6 +224,15 @@ feature {NONE} -- WEL Implementation
 			internal_wm_size_called := True
 		end
 
+
```
remove unwanted extra empty line before and after.

jvelilla replied:
Done

jocelyn commented on Src/library/vision2/interface/events/ev_dpi_action_sequence.e:
```diff
@@ -0,0 +1,60 @@
+class
+	EV_DPI_ACTION_SEQUENCE
+inherit
+	EV_ACTION_SEQUENCE [TUPLE [dpi:INTEGER; x: INTEGER; y: INTEGER; width: INTEGER; height: INTEGER]]
```
in many place, the strict style is not respected, not always critical, but for the main class interface, please add a space after the `:`

jvelilla replied:
Done

jocelyn commented on Src/library/vision2/interface/events/ev_dpi_action_sequence.e:
```diff
@@ -0,0 +1,60 @@
+			Website http://www.eiffel.com
+			Customer support http://support.eiffel.com
+		]"
+
```
remove extra empty lines

jvelilla replied:
Ok

jocelyn commented on Src/library/vision2/interface/events/ev_widget_action_sequences.e:
```diff
@@ -131,6 +131,15 @@ feature -- Event handling
+		ensure
+			not_void: Result /= Void
+		end
+
```
remove extra empty line

jvelilla replied:
Ok

jocelyn commented on Src/library/vision2/interface/support/ev_monitor_dpi_detector.e:
```diff
@@ -13,6 +13,12 @@ feature --Access
 		end
+
+	scaled_size (a_size: INTEGER): INTEGER
+			-- Return the scale size of input `a_size`.
```
comment could be shorter, as   "-- Scaled size of `a_size`. "

jvelilla replied:
Done.

jocelyn commented on Src/library/wel/windows/wel_window.e:
```diff
@@ -1419,6 +1419,15 @@ feature {WEL_WINDOW} -- Messages
+			exists: exists
+		do
+		end
+
```
remove empty line

jvelilla replied:
Ok

jocelyn commented on research/extension/autoproof/framework/eiffel2boogie/tests/framing/ownership/observer_one_to_one/output.txt:
```diff
@@ -1,54 +1,54 @@
-Successfully verified.
-======================================
-F_OOO_SUBJECT_D.register
+======================================
```
exclude this file from commit

jocelyn reviewed (changes requested):
Many feature could reuse the similar "resize" feature.
Reuse or factorize code.

jvelilla committed:
Updated class EV_MONITOR_DPI_DETECTOR, features dpi and scaled_size are now
instance-free.
Removed unneeded empty lines.
Updated code to use instace-free call to EV_MONITOR_DPI_DETECTOR features.

jocelyn commented on Src/Eiffel/interface/new_graphical/preference_tool/parameters/eb_development_window_data.e:
```diff
@@ -37,6 +40,17 @@ feature -- Element change
+	save_size_and_dpi (a_dpi, a_width, a_height: INTEGER)
+			-- Save width and height of window and dpi.
+		do
+			width := a_width
```
reuse save_size

jocelyn commented on Src/library/vision2/implementation/implementation_interface/properties/ev_positioned_i.e:
```diff
@@ -58,6 +58,12 @@ feature -- Measurement
+	dpi: INTEGER
+			-- Window dpi.
+		do
+			Result := (create {EV_MONITOR_DPI_DETECTOR_IMP}).dpi.to_integer_32
```
instance-free call ?

jvelilla committed:
Reused code whenever is possible.
Removed unneeded empty lines.
Updated code style.

jocelyn reviewed (approved):
Good for me.

jvelilla commented:
Closed merged on rev 102994
